lib/faster_s3_url/builder.rb

```diff
@@ -20,14 +20,16 @@ class Builder
 
     MAX_CACHED_SIGNING_KEYS = 5
 
-    attr_reader :bucket_name, :region, :host, :access_key_id, :session_token
+    attr_reader :bucket_name, :region, :host, :base_url, :base_path, :access_key_id, :session_token
```

Comment:
I would be tempted to make `base_url` and `base_path` private `attr_reader`s.

Making `base_url` and `base_path` public might mean supporting their behavior going forward, and they might be more an implementation detail.